tests/integration/test_sync.py

```diff
@@ -0,0 +1,60 @@
+import datetime
+import os.path
+import time
+
+import pytest
+from executor.hnswpsql import HNSWPostgresIndexer
+from jina import DocumentArray, Flow
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+compose_yml = os.path.abspath(os.path.join(cur_dir, '..', 'docker-compose.yml'))
+
+METRIC = 'cosine'
+
+
+@pytest.mark.parametrize('docker_compose', [compose_yml], indirect=['docker_compose'])
+def test_sync(docker_compose, get_documents):
+    def verify_status(f, expected_size_min):
+        result = f.post('/status', None, return_results=True)
+        result_docs = result[0].docs
+        first_hnsw_docs = sum(d.tags['hnsw_docs'] for d in result_docs)
+        psql_docs = int(result_docs[0].tags['psql_docs'])
+        assert psql_docs >= expected_size_min
+        assert int(first_hnsw_docs) >= expected_size_min
+        status = result_docs[0].tags['last_sync']
+        last_sync_timestamp = datetime.datetime.fromisoformat(status)
+        return first_hnsw_docs, last_sync_timestamp
+
+    emb_size = 10
+    nr_docs_batch = 3
+    nr_runs = 4
+
+    uses_with = {'dim': emb_size, 'sync_interval': 5}
+    search_docs = DocumentArray(
+        get_documents(index_start=nr_docs_batch * (nr_runs + 1), emb_size=emb_size)
+    )
+
+    f = Flow().add(
+        uses=HNSWPostgresIndexer,
+        uses_with=uses_with,
+    )
+
+    with f:
```

Comment:
Do we need a Flow for this? Or could we just do it on the Executor object?

Reply:
We could, but doing it in a Flow is more "realistic"

Reply:
Yes, I am OK with having an integration test, that tests with the Flow. But not each and every test must test the whole environment. Until recently we had the necessity to test on Flow level. But now this is gone and testing on lower level makes the tests less complex. This helps others understanding, what actually happens.

Reply:
> But now this is gone

Why is the necessity gone now?

Reply:
My understanding was, that until recently `MyExecutor()` would not initialize all the implicit arguments that usually would be initialized by the runtime. But now it does and thus `MyExecutor()` for local testing is a valid way to go. Might be, that my understanding here is wrong.

Reply:
Yes, you can fake them by passing runtime_args and so on as args to the `__init__`. But I'd say testing with the full Flow is the scenario we want to showcase and make sure runs as expected.